tsl/src/chunk.c

```diff
@@ -336,6 +336,24 @@ chunk_create_replica_table(PG_FUNCTION_ARGS)
 	PG_RETURN_VOID();
 }
 
+extern void
+chunk_copy_set_stage(ChunkCopyData *ccd, const char *stage)
+{
+	NameData application_name;
+
+	ccd->completed_stage = stage;
+
+	snprintf(application_name.data,
+			 sizeof(application_name.data),
+			 "%s:%s",
+			 ccd->operation_id.data,
+			 ccd->completed_stage);
+
+	pgstat_report_appname(application_name.data);
+
+	chunk_copy_activity_update(ccd);
+}
```

Comment:
Is there any case where we shouldn't commit and start a new transaction when we transition to another stage? Wondering if it makes sense to put that at the end of this function instead of repeating every time we call the function.